### PR TITLE
Add bloblang `trim_prefix` and `trim_suffix` string methods

### PR DESCRIPTION
**Pull request by mihaitodor**

### CODE REVIEW

File: internal/bloblang/query/methods_strings.go [hunk 1]
```diff
@@ -1940,3 +1940,63 @@ root.description = this.description.trim()`,
...
+root.description = this.description.trim_prefix("foobar_")`,
+			`{"description":"unchanged","name":"foobar_blobton"}`,
+			`{"description":"unchanged","name":"blobton"}`,
+		),
+	).Param(ParamString("prefix", "The leading prefix substring to trim from the string.")),
+	func(args *ParsedParams) (simpleMethod, error) {
+		prefix, err := args.FieldString("prefix")
+		if err != nil {
+			return nil, err
+		}
```
Comment (nibbleshift) [inline internal/bloblang/query/methods_strings.go:1961]: You could add byte conversion here and avoid the conversion each execution:
```
bytesPrefix = []byte(prefix)
```

File: internal/bloblang/query/methods_strings.go [hunk 2]
```diff
@@ -1940,3 +1940,63 @@ root.description = this.description.trim()`,
...
+root.description = this.description.trim_suffix("_foobar")`,
+			`{"description":"unchanged","name":"blobton_foobar"}`,
+			`{"description":"unchanged","name":"blobton"}`,
+		),
+	).Param(ParamString("suffix", "The trailing suffix substring to trim from the string.")),
+	func(args *ParsedParams) (simpleMethod, error) {
+		suffix, err := args.FieldString("suffix")
+		if err != nil {
+			return nil, err
+		}
```
Comment (nibbleshift) [inline internal/bloblang/query/methods_strings.go:1991]: You could add byte conversion here and avoid the conversion each execution:
```
bytesSuffix := []byte(suffix)
```

File: internal/bloblang/query/methods_strings.go (hunk 1 above)
Comment (mihaitodor) [reply internal/bloblang/query/methods_strings.go:1961]: Good point, thanks! ❤️

**Review by Jeffail — APPROVED** [review]
LGTM, thanks @mihaitodor!

**Pull request merged**